### PR TITLE
Zpi 198 backend endpoint game id users

### PR DESCRIPTION
**Pull request by pdrzazgaa**
Please test endpoints:
- GET games/popular
- GET game-instances/search
- GET games/{gameId}/users
- GET /game-instances

### CODE REVIEW

File: backend/src/main/java/com/zpi/backend/game/GameRepository.java [hunk 1]
```diff
@@ -54,4 +54,19 @@ public interface GameRepository extends JpaRepository<Game, Long> {
             "where GC.category_id in :categories and GS.status = 'Accepted' and lower(G.name) like %:name%",
             nativeQuery = true)
     Page<Game>  searchAllByNameContainsAndAcceptedAndCategoriesIn(@Param("name") String name, @Param("categories") List<Integer> categoriesIds, Pageable pageable);
+
+    @Query(value = "select u.uuid as uuid, u.first_name as firstName, u.last_name as lastName, " +
+            "u.location_latitude as locationLatitude, u.location_longitude as locationLongitude, " +
+            "u.avatar_link as avatarLink, " +
+            "u.avg_rating as userRate, gi.uuid as gameInstanceUUID, " +
+            "g.name as gameName, gi.avg_rating as gameInstanceRate," +
+            "sqrt(pow(u.location_latitude - :userLatitude, 2) + pow(u.location_longitude - :userLongitude, 2)) as direction " +
```
Comment (JakubSamulski) [inline backend/src/main/java/com/zpi/backend/game/GameRepository.java:63]: why direction? maybe distance?

File: backend/src/main/java/com/zpi/backend/game_instance/GameInstance.java [hunk 2]
```diff
@@ -40,5 +41,6 @@ public GameInstance(NewGameInstanceDTO newGameInstanceDTO, Game game, User owner
         this.description = newGameInstanceDTO.getDescription();
         this.isActive = true;
         this.images = new ArrayList<>();
+        this.avgRating = 0;
```
Comment (JakubSamulski) [inline backend/src/main/java/com/zpi/backend/game_instance/GameInstance.java:44]: what's that for?

File: backend/src/main/java/com/zpi/backend/game/GameService.java [hunk 3]
```diff
@@ -100,4 +105,33 @@ public void acceptGame(Authentication authentication, long id) throws GameDoesNo
...
+        Page<Object[]> objectsPage = gameRepository.getAllUsersWithGameAndRating(gameId, latitude, longitude, pageable);
+        List<UserWithGameOpinionDTO> userWithGameDTOPage =
+                objectsPage.stream()
+                        .map(this::convertToDTO)
+                        .toList();
+        return new ResultsDTO<>(userWithGameDTOPage,
+                new Pagination(objectsPage.getTotalElements(), objectsPage.getTotalPages()));
+    }
+
+    private UserWithGameOpinionDTO convertToDTO(Object[] columns) {
```
Comment (JakubSamulski) [inline backend/src/main/java/com/zpi/backend/game/GameService.java:121]: can't it be done without using plain object? it's not very readable, if not maybe add comment what is in each column

File: backend/src/main/java/com/zpi/backend/game_instance/GameInstance.java (hunk 2 above)
Comment (pdrzazgaa) [reply backend/src/main/java/com/zpi/backend/game_instance/GameInstance.java:44]: to add default value, but right, it's not nessesery.

File: backend/src/main/java/com/zpi/backend/game/GameService.java (hunk 3 above)
Comment (pdrzazgaa) [reply backend/src/main/java/com/zpi/backend/game/GameService.java:121]: I'll leave comments here. I've tried use a DTO in query, but there was a problem with converting it to custom class.

**Comment by JakubSamulski** [comment]
Also POST/game-instances does not work 
image

**Review by JakubSamulski — APPROVED** [review]

**Pull request merged**